Lib/test/test_dataclasses.py

```diff
@@ -1737,23 +1737,25 @@ class C:
                 i: int = field(metadata=0)
 
         # Make sure an empty dict works.
+        d = {}
         @dataclass
         class C:
-            i: int = field(metadata={})
-        self.assertFalse(fields(C)[0].metadata)
-        self.assertEqual(len(fields(C)[0].metadata), 0)
+            i: int = field(metadata=d)
+        self.assertEqual(fields(C)[0].metadata, d)
+        d['foo'] = 1
+        self.assertEqual(fields(C)[0].metadata, d)
         with self.assertRaisesRegex(TypeError,
                                     'does not support item assignment'):
             fields(C)[0].metadata['test'] = 3
 
         # Make sure a non-empty dict works.
+        d = {'test': 10, 'bar': '42', 3: 'three'}
         @dataclass
         class C:
-            i: int = field(metadata={'test': 10, 'bar': '42', 3: 'three'})
-        self.assertEqual(len(fields(C)[0].metadata), 3)
```

Comment:
My intent with putting explicit checks for the length and values here was to ensure that if the original mapping was changed by MappingProxy, these test would detect that. I realize it's a low probability test, but I'd like to leave it in.

Reply:
Yes, I'd like to leave it in too.

Reply:
That makes sense, I have added the tests back in bec2bad.